.github/workflows/solidity.yml

```diff
@@ -167,3 +184,71 @@ jobs:
           hostname: ${{ secrets.GRAFANA_INTERNAL_HOST }}
           this-job-name: Prettier Formatting
         continue-on-error: true
+
+  publish-beta:
+    name: Publish Beta NPM
+    environment: publish
+    needs: [initialize, changes, lint, prettier, native-compile, prepublish-test ]
+    runs-on: ubuntu-latest
+    if: needs.initialize.outputs.is-pre-release == 'true'
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Setup NodeJS
+        uses: ./.github/actions/setup-nodejs
+
+      - name: Publish to NPM
+        uses: smartcontractkit/.github/actions/ci-publish-npm@7b965cf1a0d58d05a9b8a45fffd929cd1ee18393 # feat/ci-publish-npm-0-3-0
+        with:
+          npm-token: ${{ secrets.NPM_TOKEN }}
+          create-github-release: false
+          # TODO: Remove --dry-run when ready to publish
+          publish-command: "pnpm publish-beta --dry-run --no-git-checks"
+          package-json-directory: contracts
+
+      - name: Collect Metrics
+        id: collect-gha-metrics
+        uses: smartcontractkit/push-gha-metrics-action@0281b09807758be1dcc41651e44e62b353808c47 # v2.1.0
+        with:
+          org-id: ${{ secrets.GRAFANA_INTERNAL_TENANT_ID }}
+          basic-auth: ${{ secrets.GRAFANA_INTERNAL_BASIC_AUTH }}
+          hostname: ${{ secrets.GRAFANA_INTERNAL_HOST }}
+          this-job-name: Publish Beta NPM
+        continue-on-error: true
+
+  publish-prod:
+    name: Publish Prod NPM
+    environment: publish
+    needs: [initialize, changes, lint, prettier,  native-compile, prepublish-test ]
+    runs-on: ubuntu-latest
+    permissions:
```

Comment:
why does this need `contents: write` and `publish-beta` does not?

aren't we using this for github token?

`github-token: ${{ secrets.GITHUB_TOKEN }}`

Reply:
Publish beta doesn't create a GH release, and so the permissions for the token are not set, and the token is not passed

Reply:
ahh I was asking if do we even need this `permissions` here for `publish-prod` if we are just passing in the `secrets.GITHUB_TOKEN` to the composite action?

Reply:
I believe `contents: write` is necessary for creating github releases

Reply:
oh interesting. I thought it would be take care of here?

```
    - name: Create GitHub Release
      if: inputs.create-github-release == 'true'
      shell: bash
      env:
        GITHUB_TOKEN: ${{ inputs.github-token }}
      run: |
        gh release create ${{ inputs.github-release-tag-name }} --notes "View the changelog at [${{ inputs.github-release-changelog-path }}](${{ inputs.github-release-changelog-path }})"
```

Reply:
I believe the github token is generated at the start of a job, then we pass it to this action which uses it